eppo_core/src/events/vec_event_queue.rs

```diff
@@ -14,15 +16,22 @@ pub trait EventQueue {
     /// Returns whether the queue contains enough Pending events for delivering *at least* one batch.
     fn is_batch_full(&self) -> bool;
 
+    /// Marks the provided events as failed and increments their attempts and adds it to the failed queue.
     fn mark_events_as_failed(&self, failed_events: Vec<QueuedEvent>);
 }
 
+#[derive(Debug, Clone)]
+pub struct VecEventQueueConfig {
+    pub batch_size: usize,
+    pub max_queue_size: usize,
+    pub max_retries: u8,
+}
+
 /// A simple event queue that stores events in a vector
 #[derive(Debug, Clone)]
 pub struct VecEventQueue {
-    batch_size: usize,
-    max_queue_size: usize,
-    event_queue: Arc<Mutex<HashMap<QueuedEventStatus, VecDeque<QueuedEvent>>>>,
+    config: VecEventQueueConfig,
+    event_queue: Arc<Mutex<HashMap<QueuedEventStatus, LinkedHashSet<QueuedEvent>>>>,
```

Comment:
minor: `LinkedHashSet` uses a linked list underneath and is not super-efficient for batched operations (or for memory consumption)

I know that you're trying to preserve events order but we already lost it by moving failed events to another queue (or just allowing partial failures to be re-submitted, or submitting multiple batches concurrently). So I don't think it's worth maintaining.

Looking at the code a bit more, we're still popping events from the front and always inserting to the back. I don't think `LinkedHashSet` gives us anything at all.